### ISSUE TITLE
Work sharing - inheritance issues

### ISSUE THREAD

**Issue by laritakr** [samvera/hyrax#3274]
### Descriptive summary

When creating a work which gets permissions sharing from an admin set (and optionally a collection), the inheritance does not inherit correctly when individual permissions are added on the sharing tab during creation. 

In addition, the sharing sometimes crashed, resulting in 
```
NoMethodError at /concern/generic_works
undefined method `update' for nil:NilClass
```
All subsequent transaction attempts result in Ldp::Conflict (Issue #3128)

![screen shot 2018-09-13 at 4 56 39 pm](https://user-images.githubusercontent.com/17851674/45515393-030ce200-b776-11e8-8235-35b442d43a7a.png)

### Steps to reproduce the crash behavior

**The following steps crashed the app:**
1. Started with Hyrax seed data users and collection types
2. Admin user lrobins5 created admin set. 
- Managers: mgr1, lrobins5; 
- Depositor: foo, 
- Viewers: dep2, vw1
3. Foo created Sharing type collection.
- managers: foo, mgr2;
- depositors: none
- viewers: dep2, vw1, vw2
4. Foo create work, selecting the above admin set and collection. At sharing tab, enter:
- user vw1 >> view/download access
- user vw2 >> edit access
- user mgr2 >> edit access
5. Save work. It crashes somewhere during permission issue. 

### Steps which gave incorrect permissions

Doing the above steps, but only duplicating or changing ONE permission in step 4 rather than three permissions, did not crash the app, but it created a work with lrobins5 permission missing, and had the duplicated or changed permission listed twice. 

### Correct permissions

Creating the work without adding any permissions on the sharing tab resulted in the correct permissions inherited from the admin set and collection:

![screen shot 2018-09-13 at 4 14 53 pm](https://user-images.githubusercontent.com/17851674/45515972-af02fd00-b777-11e8-9df2-5213417877ea.png)

**Comment by laritakr** [comment]
My suggestion would be to disallow individual sharing until after the work is created. I believe it is the overlapping permission options causing the problems. I didn't encounter any bugs while editing permissions on an existing work.

**Comment by rjkati** [comment]
Unable to reproduce behavior that crashed the app. This might have been fixed in #3974 and #3805. @jlhardes

**Comment by jlhardes** [comment]
Closing for now since we cannot reproduce.
